prboom2/src/gl_shader.c

```diff
@@ -53,217 +54,534 @@
 #include "r_things.h"
 #include "doomdef.h"
 
-// Indexed lighting shader uniform bindings
-typedef struct shdr_indexed_unif_s
+typedef struct shdr_def_s
 {
-  int lightlevel_index; // float
-} shdr_indexed_unif_t;
+  const GLchar* name;
+  const GLchar* value;
+} shdr_def_t;
 
-// Fuzz shader uniform bindings
-typedef struct shdr_fuzz_unif_s
+typedef struct shdr_src_s
 {
-  // (vec2) sprite texture dimensions
-  int tex_d_index;
-  // (float) ratio of screen resolution to fuzz resolution
-  int ratio_index;
-  // (float) random seed
-  int seed_index;
-} shdr_fuzz_unif_t;
+  GLchar const** strs;
+  GLint* lens;
+  unsigned int size;
+  unsigned int cap;
+} shdr_src_t;
 
-static GLShader *sh_indexed = NULL;
-static shdr_indexed_unif_t indexed_unifs;
-static GLShader *sh_fuzz = NULL;
-static shdr_fuzz_unif_t fuzz_unifs;
-static GLShader *active_shader = NULL;
-
-static GLShader* gld_LoadShader(const char *vpname, const char *fpname);
+static void ShaderSrcInit(shdr_src_t* src)

```

Comment:
Are you using a different capitalization because it's static? I experimented with this at some point but ultimately decided I prefer the `prefix_FunctionName` style across the board.

Reply:
Should these be `gld` or `glsl` functions?  The file already contains both.